src/rgw/rgw_op.h

```diff
@@ -1485,6 +1485,29 @@ class RGWDeleteObj : public RGWOp {
   dmc::client_id dmclock_client() override { return dmc::client_id::data; }
 };
 
+enum class TaggingPermission : short {
+NO_TAGGING_PERM,
+PUT_TAGGING_PERM,
+GET_TAGGING_PERM
+};
+
+enum class TaggingDirective {
+NO_TAGGING_DIRECTIVE,
+COPY_TAGGING_DIRECTIVE,
+REPLACE_TAGGING_DIRECTIVE
```

Comment:
with enum class, calling code has to refer to these values like `TaggingDirective::NO_TAGGING_DIRECTIVE`, so the `_TAGGING_DIRECTIVE` suffix is redundant